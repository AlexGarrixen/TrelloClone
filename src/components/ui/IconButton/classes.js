const classes = {
  default: 'icon-button',
  // class applied if variant="contained" and color="default"
  containedDefault: 'button--contained-default',
  // class applied if variant="text" and color="default"
  textDefault: 'button--text-default',
  // class applied if variant="outlined" and color="default"
  outlinedDefault: 'button--outlined-default',
  // class applied if variant="contained" and color="primary"
  containedPrimary: 'button--contained-primary',
  // class applied if variant="text" and color="primary"
  textPrimary: 'button--text-primary',
  // class applied if variant="outlined" and color="primary"
  outlinedPrimary: 'button--outlined-primary',
  // class applied if variant="texcontained" and color="secondary"
  containedSecondary: 'button--contained-secondary',
  // class applied if variant="text" and color="secondary"
  textSecondary: 'button--text-secondary',
  // class applied if variant="outlined" and color="secondary"
  outlinedSecondary: 'button--outlined-secondary',
  // class apploed if disabled=true
  disabled: 'disabled',
};

export default classes;
