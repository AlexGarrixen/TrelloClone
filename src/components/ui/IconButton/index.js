import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import capitalize from '../../../utils/capitalize';
import classes from './classes';

const IconButton = forwardRef(
  (
    {
      children,
      component: Component = 'button',
      className,
      color = 'default',
      variant = 'contained',
      type = 'button',
      ...other
    },
    ref
  ) => (
    <Component
      {...other}
      ref={ref}
      type={type}
      className={clsx(
        classes.default,
        classes[`${variant}${capitalize(color)}`],
        className
      )}
    >
      {children}
    </Component>
  )
);

IconButton.propTypes = {
  children: PropTypes.node,
  component: PropTypes.elementType,
  color: PropTypes.oneOf(['default', 'primary', 'secondary']),
  variant: PropTypes.oneOf(['text', 'contained', 'outlined']),
  type: PropTypes.string,
};

export default IconButton;
