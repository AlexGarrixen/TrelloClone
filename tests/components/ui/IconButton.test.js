import React from 'react';
import { mount } from 'enzyme';
import IconButton from '../../../src/components/ui/IconButton';

describe('<IconButton />', () => {
  test('Render success', () => {
    const wrapper = mount(
      <IconButton>
        <svg />
      </IconButton>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('svg').exists()).toBe(true);
  });

  test('Test with default props', () => {
    const wrapper = mount(
      <IconButton>
        <svg />
      </IconButton>
    );
    const element = wrapper.find('.icon-button');

    expect(element.type()).toBe('button');
    expect(element.find('svg').exists()).toBe(true);
    expect(element.hasClass('button--contained-default')).toBe(true);
  });

  test('Test with props variant and color', () => {
    const wrapper = mount(
      <IconButton variant='outlined' color='primary'>
        <svg />
      </IconButton>
    );
    const element = wrapper.find('.icon-button');

    expect(element.hasClass('button--outlined-primary')).toBe(true);
  });

  test('Test with prop component', () => {
    const wrapper = mount(
      <IconButton component='a'>
        <svg />
      </IconButton>
    );
    const element = wrapper.find('.icon-button');

    expect(element.type()).toBe('a');
  });
});
