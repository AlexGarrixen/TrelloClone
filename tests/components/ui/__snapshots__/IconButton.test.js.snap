// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> Render success 1`] = `
<button
  className="icon-button button--contained-default"
  type="button"
>
  <svg />
</button>
`;
